core/src/main/java/org/apache/iceberg/rest/RESTCatalog.java

```diff
@@ -55,7 +59,50 @@ public class RESTCatalog
   public RESTCatalog() {
     this(
         SessionCatalog.SessionContext.createEmpty(),
-        config -> HTTPClient.builder(config).uri(config.get(CatalogProperties.URI)).build());
+        config -> {
+          HTTPClient.Builder builder =
```

Comment:
so I think one possible issue that I can currently think of is that this approach will only work for the REST catalog itself, but things like refreshing vended credentials with S3/GCS (`VendedCredentialsProvider`/`OAuth2RefreshCredentialsHandler`)or S3 signing (`S3V4RestSignerClient`) won't work, since those places instantiate their own HTTP client that wouldn't configure the proxy